File: verify_users.py
import time
import requests
from playwright.sync_api import sync_playwright

def seed_user():
    url = "http://localhost:50050/api/v1/users"
    headers = {"X-API-Key": "test-token"}
    user = {
        "id": "verify-admin",
        "authentication": {
            "basic_auth": {
                "username": "verify-admin",
                # hash for "password"
                "password_hash": "$2a$12$KPRtQETm7XKJP/L6FjYYxuCFpTK/oRs7v9U6hWx9XFnWy6UuDqK/a"
            }
        },
        "roles": ["admin"]
    }
    try:
        requests.post(url, json={"user": user}, headers=headers)
        print("Seeded user: verify-admin")
    except Exception as e:
        print(f"Failed to seed user: {e}")

def run(playwright):
    browser = playwright.chromium.launch()
    context = browser.new_context()
    page = context.new_page()

    # 1. Login
    page.goto("http://localhost:9002/login")
    page.fill('input[name="username"]', 'verify-admin')
    page.fill('input[name="password"]', 'password')
    page.click('button[type="submit"]')
    page.wait_for_url("http://localhost:9002/")

    # 2. Go to Users
    page.goto("http://localhost:9002/users")

    # 3. Open Add User Sheet
    page.click('button:has-text("Add User")')

    # 4. Fill some data to show form
    page.fill('input[name="id"]', 'new-user')

    # 5. Take screenshot
    page.screenshot(path="/home/jules/verification/verification.png")
    print("Screenshot saved to /home/jules/verification/verification.png")

    browser.close()

if __name__ == "__main__":
    seed_user()
    with sync_playwright() as playwright:
        run(playwright)

File: verify_users.js
const { chromium } = require('playwright');

async function seedUser() {
    const url = "http://localhost:50050/api/v1/users";
    const headers = { "X-API-Key": "test-token", "Content-Type": "application/json" };
    const user = {
        id: "verify-admin",
        authentication: {
            basic_auth: {
                username: "verify-admin",
                password_hash: "$2a$12$KPRtQETm7XKJP/L6FjYYxuCFpTK/oRs7v9U6hWx9XFnWy6UuDqK/a"
            }
        },
        roles: ["admin"]
    };
    try {
        const res = await fetch(url, { method: 'POST', headers, body: JSON.stringify({ user }) });
        if (res.ok) console.log("Seeded user: verify-admin");
        else console.log("Failed to seed user:", res.status, await res.text());
    } catch (e) {
        console.log("Failed to seed user:", e);
    }
}

async function run() {
    await seedUser();
    const browser = await chromium.launch();
    const context = await browser.newContext();
    const page = await context.newPage();

    try {
        // 1. Login
        await page.goto("http://localhost:9002/login");
        await page.fill('input[name="username"]', 'verify-admin');
        await page.fill('input[name="password"]', 'password');
        await page.click('button[type="submit"]');
        await page.waitForURL("http://localhost:9002/", { timeout: 15000 });

        // 2. Go to Users
        await page.goto("http://localhost:9002/users");

        // 3. Open Add User Sheet
        await page.click('button:has-text("Add User")');

        // 4. Fill some data to show form
        await page.fill('input[name="id"]', 'new-user');

        // 5. Take screenshot
        await page.screenshot({ path: "/home/jules/verification/verification.png" });
        console.log("Screenshot saved to /home/jules/verification/verification.png");
    } catch (e) {
        console.error(e);
    } finally {
        await browser.close();
    }
}

run();
